pytensor/tensor/extra_ops.py

```diff
@@ -592,6 +593,19 @@ def squeeze(x, axis=None):
         # Nothing to do
         return _x
 
+    if _x.ndim == 0:
+        # Nothing could be squeezed
+        return _x
+
+    # `Dimshuffle` raises when we try to drop an axis that is not statically broadcastable.
+    # We add a `specify_broadcastable` instead of raising.
+    non_broadcastable_axis = [i for i in axis if not _x.broadcastable[i]]
+
+    if non_broadcastable_axis:
+        if not isinstance(non_broadcastable_axis, Collection):
+            non_broadcastable_axis = (non_broadcastable_axis,)
+        _x = specify_broadcastable(_x, *non_broadcastable_axis)
```

Comment:
I think this suffices. The helper already has a check not to do anything if there are no axes
```suggestion
   _x = specify_broadcastable(_x, *non_broadcastable_axis)
```